src/cudamatrix/cu-rand-speed-test.cc

```diff
@@ -56,63 +56,166 @@ std::string MeanVariance(const CuMatrixBase<Real>& m) {
+  Timer t;
+  CuRand<Real> rand;
+  CuMatrix<Real> m(249,1001, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
     rand.RandGaussian(&m);
   }
-  KALDI_LOG << __func__ << NameOf<Real>() << " t = " << t.Elapsed() << "s, " << MeanVariance(m);
+  CuMatrix<Real> m2(256,1024, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandGaussian(&m2);
+  }
+  // flops = number of generated random numbers per second,
+  Real flops = iter * (m.NumRows() * m.NumCols() + m2.NumRows() * m2.NumCols()) / t.Elapsed();
+  KALDI_LOG << __func__ << NameOf<Real>()
+            << " Speed was " << flops << " rand_elems/s. "
+            << "(debug " << MeanVariance(m) << ")";
 }
 
 template <typename Real>
-void CuRandGaussianVectorSpeedTest() {
+void CuRandGaussianMatrixBaseSpeedTest(const int32 iter) {
   Timer t;
   CuRand<Real> rand;
-  CuVector<Real> v(2011);
-  for (int32 i = 0; i < 200; i++) {
+  CuMatrix<Real> m(249,1001, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandGaussian(dynamic_cast<CuMatrixBase<Real>*>(&m));
+  }
+  CuMatrix<Real> m2(256,1024, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandGaussian(dynamic_cast<CuMatrixBase<Real>*>(&m2));
+  }
+  // flops = number of generated random numbers per second,
+  Real flops = iter * (m.NumRows() * m.NumCols() + m2.NumRows() * m2.NumCols()) / t.Elapsed();
+  KALDI_LOG << __func__ << NameOf<Real>()
+            << " Speed was " << flops << " rand_elems/s. "
+            << "(debug " << MeanVariance(m) << ")";
+}
+
+template <typename Real>
+void CuRandUniformVectorSpeedTest(const int32 iter) {
+  Timer t;
+  CuRand<Real> rand;
+  CuVector<Real> v(2011, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandUniform(&v);
+  }
+  CuVector<Real> v2(2048, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandUniform(&v2);
+  }
+  // flops = number of generated random numbers per second,
+  Real flops = iter * (v.Dim() + v2.Dim()) / t.Elapsed();
+  KALDI_LOG << __func__ << NameOf<Real>()
+            << " Speed was " << flops << " rand_elems/s. "
+            << "(debug " << MeanVariance(v) << ")";
+}
+
+template <typename Real>
+void CuRandGaussianVectorSpeedTest(const int32 iter) {
+  Timer t;
+  CuRand<Real> rand;
+  CuVector<Real> v(2011, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
     rand.RandGaussian(&v);
   }
-  KALDI_LOG << __func__ << NameOf<Real>() << " t = " << t.Elapsed() << "s";
+  CuVector<Real> v2(2048, kUndefined);
+  for (int32 i = 0; i < iter; i++) {
+    rand.RandGaussian(&v2);
+  }
+  // flops = number of generated random numbers per second,
+  Real flops = iter * (v.Dim() + v2.Dim()) / t.Elapsed();
+  KALDI_LOG << __func__ << NameOf<Real>()
+            << " Speed was " << flops << " rand_elems/s. "
+            << "(debug " << MeanVariance(v) << ")";
 }
 
 } // namespace kaldi
 
 
 int main() {
-  for (int32 loop = 0; loop < 2; loop++) {
+  int32 iter = 10; // Be quick on CPU,
```

Comment:
As for "the other tests in cudamatrix/", do you mean I should modify all the other cu-*-test.cc files? But I think that these test files only run the test once if no GPU. So I'm not sure what do you mean by "just do an #ifdef that will ensure the loop only runs once if no GPU".